compiler/resolution/functionResolution.cpp

```diff
@@ -4675,7 +4675,15 @@ static void resolveMaybeSyncSingleField(CallExpr* call) {
     DefExpr*   tmpDefn   = new DefExpr(tmp);
 
     // Applies a type to TMP
-    CallExpr*  tmpExpr   = new CallExpr(PRIM_INIT, fieldDef->exprType->copy());
+    // BHARSH TODO: Why can't we just use the exprType of the formal? Shouldn't
+    // it be the same?
```

Comment:
We can't have the same AST inserted in two places in the code.  It shouldn't particularly matter whether we copy from the field definition or the argument we just made, though.

Reply:
I quickly noticed that the corresponding formal is not readily available, so I'll leave this as a future TODO.

Reply:
Okay, cool.  I think this is good to go, then!